secrethub_wrapper.go

```diff
@@ -0,0 +1,187 @@
+typedef long long time;
+typedef char* uuid;
+
+struct Secret {
+	uuid SecretID;
+	uuid DirID;
+	uuid RepoID;
+	char* Name;
+	char* BlindName;
+	int VersionCount;
+	int LatestVersion;
+	char* Status;
+	time CreatedAt;
+};
+
+struct SecretVersion {
+	uuid SecretVersionID;
+	struct Secret Secret;
+	int Version;
+	char* Data;
+	time CreatedAt;
+	char* Status;
+};
+#include <stdbool.h>
+*/
+import "C"
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+)
+
+// Client creates and returns a new Go client to be used by xgo clients.
+func Client() (*secrethub.Client, error) {
+	options := []secrethub.ClientOption{
+		secrethub.WithAppInfo(&secrethub.AppInfo{
+			Name:    "secrethub-xgo",
+			Version: "0.1.0",
+		}),
+	}
+	client, err := secrethub.NewClient(options...)
+	return client, err
+}
+
+// Read retrieves a secret by its path.
+//export Read
+func Read(path *C.char, errMessage **C.char) C.struct_SecretVersion {
+	client, err := Client()
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return C.struct_SecretVersion{}
+	}
+	secret, err := client.Secrets().Read(C.GoString(path))
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return C.struct_SecretVersion{}
+	}
+	return C.struct_SecretVersion{
+		SecretVersionID: C.CString(secret.SecretVersionID.String()),
+		Secret: C.struct_Secret{
+			SecretID:      C.CString(secret.Secret.SecretID.String()),
+			DirID:         C.CString(secret.Secret.DirID.String()),
+			RepoID:        C.CString(secret.Secret.RepoID.String()),
+			Name:          C.CString(secret.Secret.Name),
+			BlindName:     C.CString(secret.Secret.BlindName),
+			VersionCount:  C.int(secret.Secret.VersionCount),
+			LatestVersion: C.int(secret.Secret.LatestVersion),
+			Status:        C.CString(secret.Secret.Status),
+			CreatedAt:     C.longlong(secret.Secret.CreatedAt.Unix()),
+		},
+		Version:   C.int(secret.Version),
+		Data:      C.CString(string(secret.Data)),
+		CreatedAt: C.longlong(secret.CreatedAt.Unix()),
+		Status:    C.CString(secret.Status),
+	}
+}
+
+// ReadString retrieves a secret as a string.
+//export ReadString
+func ReadString(path *C.char, errMessage **C.char) *C.char {
+	client, err := Client()
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return nil
+	}
+	secret, err := client.Secrets().Read(C.GoString(path))
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return nil
+	}
+	return C.CString(string(secret.Data))
+}
+
+// Resolve fetches the values of a secret from SecretHub, when the `ref` parameter
+// has the format `secrethub://<path>`. Otherwise it returns `ref` unchanged, as an array of bytes.
+//export Resolve
+func Resolve(ref *C.char, errMessage **C.char) *C.char {
+	client, err := Client()
+	if err != nil {
+		*errMessage = C.CString(err.Error())
+		return nil
+	}
+	bits := strings.Split(C.GoString(ref), "://")
+	if len(bits) == 2 && bits[0] == "secrethub" {
+		secret, err := client.Secrets().Read(bits[1])
```

Comment:
Go has some handy helper functions for prefixes :grin: 
```suggestion
	lowercaseRef := strings.ToLower(C.GoString(ref))
	prefix := "secrethub://"
	if strings.HasPrefix(lowercaseRef, prefix) {
		secret, err := client.Secrets().Read(strings.TrimPrefix(lowercaseRef, prefix))
```

_Haven't compiled this, might contain some typos._

Reply:
Nice thing to see. Adjusted to the suggested implementation.